### PR TITLE
Create endpoints for managing access rights

### PR DESCRIPTION
**Pull request by Victoradukwu**
#### What does this PR do?
	add endpoints for managing access rights
#### Description of Task to be completed?
- Multiple permissions with the same name
- add get_all_permissions end point
- add update_permission end point
- add delete_permission endpoint
- fix the get_user_roles endpoint
- fix create_user_role avoid duplication
- fix create_user_role avoid non-existing roles
- Write tests for the endpoints
#### How should this be manually tested?
- Clone the branch `ft-user-roles-permission-endpoints`
- Checkout to the branch
- Start the application
- with postman test the affected endpoints
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)
￼N/A
#### Questions:
N/A

### CODE REVIEW

File: tests/integration/endpoints/test_permission_endpoints.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		role1 = RoleFactory.create(name='admin')
+		permission_repo.new_permission(role1.id, 'name-1', 'keyword-1')
+		user_id = BaseTestCase.user_id()
+		PermissionFactory.create(keyword='delete_permissions', role_id=role1.id)
+		UserRoleFactory.create(user_id=user_id, role_id=role1.id)
+		response = self.client().delete(self.make_url(f'/roles/permissions/576'), headers=self.headers())
+		response_json = self.decode_from_json_string(response.data.decode('utf-8'))
+
+		self.assert404(response)
+		self.assertEqual(response_json['msg'], 'Invalid or incorrect permission id provided')
```
Comment (Maxcutex) [inline tests/integration/endpoints/test_permission_endpoints.py]: You didnt add a test to check if the person performing the action is permitted to do so or is there no such scenerio?

Comment (Victoradukwu) [reply tests/integration/endpoints/test_permission_endpoints.py]: @Maxcutex That is exactly what the test function on line **92** is doing.

Comment (Maxcutex) [reply tests/integration/endpoints/test_permission_endpoints.py]: @Victoradukwu you only did for delete , you didnt do for update, create & list

Comment (Victoradukwu) [reply tests/integration/endpoints/test_permission_endpoints.py]: @Maxcutex  I am working on that now.

Comment (Victoradukwu) [reply tests/integration/endpoints/test_permission_endpoints.py]: @Maxcutex It's all fixed.

Comment (Maxcutex) [reply tests/integration/endpoints/test_permission_endpoints.py]: Thanks boss

**Pull request merged**